plugins/filetrees/yazi.nix

```diff
@@ -0,0 +1,161 @@
+
+        Defaults to simply opening the file.
+      '';
+    };
+
+    clipboard_register = defaultNullOpts.mkStr "*" ''
+      Some yazi.nvim commands copy text to the clipboard. This is the register
+      yazi.nvim should use for copying. Defaults to "*", the system clipboard.
+    '';
+
+    keymaps =
+      defaultNullOpts.mkNullable (types.either types.attrs (types.enum [ false ]))
+        {
+          show_help = "<f1>";
+          open_file_in_vertical_split = "<c-v>";
+          open_file_in_horizontal_split = "<c-x>";
+          open_file_in_tab = "<c-t>";
+          grep_in_directory = "<c-s>";
+          replace_in_directory = "<c-g>";
+          cycle_open_buffers = "<tab>";
+          copy_relative_path_to_selected_files = "<c-y>";
+          send_to_quickfix_list = "<c-q>";
+        }
+        ''
+          Customize the keymaps that are active when yazi is open and focused.
+
+          Also:
+            - use e.g. `open_file_in_tab = false` to disable a keymap
+            - you can customize only some of the keymaps if you want
+            - Set to `false` to disable all default keymaps.
+        '';
+
+    set_keymappings_function = defaultNullOpts.mkLuaFn null ''
+      Completely override the keymappings for yazi. This function will be
+      called in the context of the yazi terminal buffer.
+    '';
+
+    hooks = {
+      yazi_opened = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(preselected_path, yazi_buffer_id, config)
+          end
+        '';
+        description = ''
+          If you want to execute a custom action when yazi has been opened,
+          you can define it here.
+        '';
+      };
+
+      yazi_closed_successfully = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(chosen_file, config, state)
+          end
+        '';
+        description = "When yazi was successfully closed";
+      };
+
+      yazi_opened_multiple_files = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(chosen_files)
+            vim.cmd("args" .. table.concat(chosen_files, " "))
+          end
+        '';
+        description = ''
+          When yazi opened multiple files. The default is to send them to the
+          quickfix list, but if you want to change that, you can define it here
+        '';
+      };
+    };
+
+    highlight_groups = defaultNullOpts.mkAttributeSet { hovered_buffer = null; } ''
+      Add highlight groups to different yazi events.
+
+      NOTE: this only works if `use_ya_for_events_reading` is enabled, etc.
+    '';
+
+    floating_window_scaling_factor =
+      defaultNullOpts.mkNum 0.9
+        "The floating window scaling factor. 1 means 100%, 0.9 means 90%, etc.";
+
+    yazi_floating_window_winblend = defaultNullOpts.mkNullableWithRaw (types.ints.between 0
+      100
+    ) 0 "`0` for fully opaque and `100` for fully transparent. See :h winblend";
```

Comment:
Nixfmt has done a _horrible_ job here!

Reply:
Yeah... I didn't know if this was another situation where we needed to use `'` and set the options directly.

Reply:
```nix
    yazi_floating_window_winblend = defaultNullOpts.mkNullableWithRaw' {
      type = (types.ints.between 0 100);
      pluginDefault = 0;
      description = "`0` for fully opaque and `100` for fully transparent. See :h winblend";
    };
```